src/app/hooks/api/useUTXOs.ts

```diff
@@ -6,14 +6,20 @@ import { filterOrdinals } from "@/utils/utxo";
 export const UTXO_KEY = "UTXO";
 
 export function useUTXOs({ enabled = true }: { enabled?: boolean } = {}) {
-  const { getUtxos, getInscriptions, address } = useBTCWallet();
+  const { getUtxos, getInscriptions, address, shouldFilterOrdinals } =
+    useBTCWallet();
 
   const fetchAvailableUTXOs = async () => {
     if (!getUtxos || !address) {
       return;
     }
 
     const mempoolUTXOs = await getUtxos(address);
+    // Return UTXOs without filtering if not required
+    if (!shouldFilterOrdinals) {
+      return mempoolUTXOs;
+    }
+
     const filteredUTXOs = await filterOrdinals(
       mempoolUTXOs,
       address,
```

Comment:
useUTXOs should be only responsible for fetching data (not filtering it). `filderOrdinals` method is a legacy